improver/nowcasting/forecasting.py

```diff
@@ -222,6 +226,90 @@ def point_in_bounds(x, y, nx, ny):
+                Time coordinate from source cube
+            advected_cube (iris.cube.Cube):
+                Cube containing advected data (modified in place)
+            timestep (datetime.timedelta)
+                Time difference between the advected output and the source
+        """
+        original_datetime, = (input_time.units).num2date(input_time.points)
+        new_datetime = original_datetime + timestep
+        new_time = (input_time.units).date2num(new_datetime)
+        advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(TIME_REFERENCE_UNIT)
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(
+                TIME_REFERENCE_DTYPE))
+
+    @staticmethod
+    def _add_forecast_reference_time(input_time, advected_cube):
+        """Add or replace a forecast reference time on the advected cube"""
+        try:
+            advected_cube.remove_coord("forecast_reference_time")
+        except CoordinateNotFoundError:
+            pass
+
+        frt_coord = input_time.copy()
+        frt_coord.rename("forecast_reference_time")
+        frt_coord.convert_units(TIME_REFERENCE_UNIT)
+        frt_coord.points = np.around(frt_coord.points).astype(
+            TIME_REFERENCE_DTYPE)
+        advected_cube.add_aux_coord(frt_coord)
+
+    @staticmethod
+    def _add_forecast_period(advected_cube, timestep):
+        """Add or replace a forecast period on the advected cube"""
+        try:
+            advected_cube.remove_coord("forecast_period")
+        except CoordinateNotFoundError:
+            pass
+
+        forecast_period_seconds = np.int32(timestep.total_seconds())
+        forecast_period_coord = AuxCoord(forecast_period_seconds,
+                                         standard_name="forecast_period",
+                                         units="seconds")
+        advected_cube.add_aux_coord(forecast_period_coord)
+
+    def _create_output_cube(self, cube, advected_data, timestep):
+        """
+        Create a cube and appropriate metadata to contain the advected forecast
+
+        Args:
+            cube (iris.cube.Cube):
+                Source cube (before advection)
+            advected_data (numpy.ndarray):
+                Advected data
+            timestep (datetime.timedelta):
+                Time difference between the advected output and the source
+
+        Returns:
+            iris.cube.Cube
+        """
+        attributes = generate_mandatory_attributes([cube])
+        if "institution" in cube.attributes.keys():
+            attributes["source"] = "{} Nowcast".format(
+                attributes["institution"])
+        else:
+            attributes["source"] = "Nowcast"
+        advected_cube = create_new_diagnostic_cube(
+            cube.name(), cube.units, cube, attributes, data=advected_data)
+        amend_attributes(advected_cube, self.attributes_dict)
```

Comment:
Yes.  We need to set up new attributes for source, institution, model ID and temperature for the nowcast, because it is technically a new model.  However we need this to be flexible for different institutions.  A few solutions to this have been explored in previous PRs, including having a module where different NMS's metadata could be specified in the code base.  However the prevailing attitude was that we should avoid having those attributes coded into IMPROVER and preserve the ability to set them from the command line in the specific case of nowcasting.